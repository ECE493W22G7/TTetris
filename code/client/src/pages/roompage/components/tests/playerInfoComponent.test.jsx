
import React from 'react';
import renderer from 'react-test-renderer';
import PlayerInfoComponent from '../playerInfoComponent';
import SocketMock  from 'socket.io-mock';
import { toMatchDiffSnapshot } from 'snapshot-diff';
const {act} = renderer;

expect.extend({ toMatchDiffSnapshot });

let createClient = (id, nick, color) => {
    return {
        "id": id,
        "nickname": nick,
        "color": color
    }
}

function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

describe("PlayerInfoComponent", () => {
    let socket;
    let c1, c2; 

    beforeEach(() => {
        c1 = createClient(1, "test","#000");
        c2 = createClient(2, "test2","#f00");
        socket = new SocketMock();
        socket.on("getConnectedClients", (roomId, done) => {
            done([c1, c2]);
        })
    })
    test('renders correctly', async () => {
        let component;
        act(() => {
            component = renderer.create(<PlayerInfoComponent socket={socket.socketClient} roomID={"34223"} />);
        });

        await sleep(2000);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("should show player nick name when a socket event is received", async () => {
        let component;
        act(() => {
            component = renderer.create(<PlayerInfoComponent socket={socket.socketClient} roomID={"34223"} />);

        });
        
        let oldTree = component.toJSON();

        // server informs us of player change
        c1.nickname = "newName";
        act(() => {
            socket.emit("connectClient", [c1, c2]);
        })
        
        let newTree = component.toJSON();
        expect(oldTree).toMatchDiffSnapshot(newTree);
    })

})

