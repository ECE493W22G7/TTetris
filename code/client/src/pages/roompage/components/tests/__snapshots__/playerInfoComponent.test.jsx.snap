// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInfoComponent renders correctly 1`] = `
<div
  className="players-content card"
>
  <link
    href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.7.2/font/bootstrap-icons.css"
    rel="stylesheet"
  />
  <div
    className="player-title font-italic text-center card-header"
  >
    Players Info
  </div>
  <div
    className="player-list"
  >
    <div
      className="player-info"
    >
      <div
        className="player-name-button"
      >
        <span
          className=" text-left font-weight-bold"
        >
          test
        </span>
        <button
          className="btn btn-link"
          type="button"
        >
          <span
            className="bi bi-mic fa-lg"
          />
        </button>
      </div>
      <div
        className="piece-container"
      >
        <div
          className="player-piece"
          style={
            Object {
              "backgroundColor": "#000",
            }
          }
        />
        <div
          className="player-emergency"
        />
      </div>
    </div>
    <div
      className="player-info"
    >
      <div
        className="player-name-button"
      >
        <span
          className=" text-left font-weight-bold"
        >
          test2
        </span>
        <button
          className="btn btn-link"
          type="button"
        >
          <span
            className="bi bi-mic fa-lg"
          />
        </button>
      </div>
      <div
        className="piece-container"
      >
        <div
          className="player-piece"
          style={
            Object {
              "backgroundColor": "#f00",
            }
          }
        />
        <div
          className="player-emergency"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`PlayerInfoComponent should show player color when a socket event is received 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -38,11 +38,11 @@
        >
          <div
            className=\\"player-piece\\"
            style={
              Object {
-               \\"backgroundColor\\": \\"#000\\",
+               \\"backgroundColor\\": \\"#ff0\\",
              }
            }
          />
          <div
            className=\\"player-emergency\\"
@@ -74,11 +74,11 @@
        >
          <div
            className=\\"player-piece\\"
            style={
              Object {
-               \\"backgroundColor\\": \\"#f00\\",
+               \\"backgroundColor\\": \\"#000\\",
              }
            }
          />
          <div
            className=\\"player-emergency\\""
`;

exports[`PlayerInfoComponent should show player nick name when a socket event is received 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -20,11 +20,11 @@
          className=\\"player-name-button\\"
        >
          <span
            className=\\" text-left font-weight-bold\\"
          >
-           test
+           newName
          </span>
          <button
            className=\\"btn btn-link\\"
            type=\\"button\\"
          >"
`;
